src/main/kotlin/org/rust/ide/inspections/lints/RsPathUsageAnalysis.kt

```diff
@@ -78,6 +78,10 @@ private fun handleElement(
             }
             true
         }
+        is RsMacroCall -> {
+            handleElement(element.path, directUsages, traitUsages)
+            true
```

Comment:
```suggestion
```
It's probably overkill, but for future-proofing this should maybe return the same thing as the recursive call?

Reply:
If we return false here, the entire macro expansion will be skipped, and return value of `handleElement` indicated whether or not we should go inside `element.path` subtree. It turns out my implementation doesn't handle qualified paths, I fixed it now, thanks!

Reply:
Hmm. But I wonder why this change was needed though? In e.g. `foo::bar!`, the root path `foo::bar` shouldn't have a qualifier, so it should be handled by the `else` branch at :71, which should mark `foo` as visited path.

Reply:
The root path `foo::bar` has a qualifier `foo`, isn't it?

Reply:
Oh, nevermind, I again confused the order of nested paths produced by the parser. It confuses me that the qualifier is on the left, but at the same time it is the child of the root path. Thanks for the explanation.